### PR TITLE
E UI 4708 hearing venue service location

### PR DESCRIPTION
**Pull request by waynepHMCTS**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-4708

### Change description ###
Consumption off Service location  with tag 1.4.9-search-location-component-common

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

**Review by kuda1992 — APPROVED** [review]
Looks goof

File: src/cases/containers/case-hearings/case-hearings.component.html [hunk 1]
```diff
@@ -6,7 +6,7 @@
      class="govuk-button
      govuk-button--start"
      data-module="govuk-button"
-     [routerLink]="['/', 'hearings', 'request']"
+     [routerLink]="['/', 'hearings', 'request', 'locationSearch']"
```
Comment (rayliangatos) [inline src/cases/containers/case-hearings/case-hearings.component.html:9]: Can we revert this and ask QA to test on:
/hearings/request/locationSearch url directly?

File: src/hearings/containers/request-hearing/location-search-container/location-search-container.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
+import { provideMockStore } from '@ngrx/store/testing';
+import { LocationSearchContainerComponent } from './location-search-container.component';
+
+fdescribe('LocationSearchContainerComponent', () => {
```
Comment (rayliangatos) [inline src/hearings/containers/request-hearing/location-search-container/location-search-container.component.spec.ts:6]: fdescribe

File: src/hearings/containers/request-hearing/location-search-container/location-search-container.component.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+export class LocationSearchContainerComponent implements OnInit {
+  public serviceId: string = 'SSCS';
+
+  constructor(private readonly hearingStore: Store<fromHearingStore.State>) {}
+
+  public ngOnInit(): void {
+    this.hearingStore.pipe(select(fromHearingStore.getHearingsList)).pipe(
+      map(hearingList => hearingList.caseHearingsMainModel ? hearingList.caseHearingsMainModel.hmctsServiceID : '')
+    ).subscribe(id => {
+      this.serviceId = id
```
Comment (rayliangatos) [inline src/hearings/containers/request-hearing/location-search-container/location-search-container.component.ts:19]: missed ;

File: src/hearings/containers/request-hearing/request-hearing.component.html [hunk 4]
```diff
@@ -2,9 +2,9 @@
 <div class="govuk-width-container">
   <div class="page">
     <main id="content" role="main">
-      <p>
-        request-hearing works!
-      </p>
+        <div class="search-location">
+          <exui-search-location class="search-location"></exui-search-location>
+        </div>
```
Comment (rayliangatos) [inline src/hearings/containers/request-hearing/request-hearing.component.html:7]: @Kartheek-S has a pr https://github.com/hmcts/rpx-xui-webapp/pull/1481 which should be merged first, so your pr should depends on his, and this changes is not necessary anymore if we use parent route. As this seems duplicated with location-search-container.component.html

File: src/hearings/hearings.module.ts [hunk 5]
```diff
@@ -24,7 +25,10 @@ import { effects, reducers } from './store';
     SharedModule,
     FormsModule,
     ReactiveFormsModule,
-    ExuiCommonLibModule
+    ExuiCommonLibModule,
+    MatAutocompleteModule,
+    MatTabsModule,
+    MatInputModule
```
Comment (rayliangatos) [inline src/hearings/hearings.module.ts:31]: You probably don't need this Mat** in the hearings module, as we assume all the dependencies should be in common-lib module.

**Review by rayliangatos — CHANGES_REQUESTED** [review]

File: package.json [hunk 6]
```diff
@@ -20,7 +20,7 @@
     "coverage:node": "cd api && yarn coverage",
     "ng": "ng",
     "start:ng": "ng serve --port 3000 --host 0.0.0.0 --proxy-config proxy.config.json",
-    "build:ng": "ng build",
+    "build:ng": "node --max_old_space_size=8192 node_modules/@angular/cli/bin/ng build --prod build",
```
[the remark this answers is not in the document]
Comment (waynepHMCTS) [reply package.json:23]: memory problem

File: src/cases/case-feature.module.spec.ts [hunk 7]
```diff
@@ -1,13 +1,116 @@
...
-  it('should create an instance', () => {
-    expect(caseFeatureModule).toBeTruthy();
+  it('should hasReadOnlyAction if status upcoming', () => {
+    component.status = EXUISectionStatusEnum.UPCOMING;
+    component.actions = [Actions.READ];
+    component.ngOnInit();
+    fixture.detectChanges();
+    expect(component.hasReadOnlyAction).toBeTruthy();
   });
 });
```
[the remark this answers is not in the document]
Comment (waynepHMCTS) [reply src/cases/case-feature.module.spec.ts:116]: This have been merged in by the feature branch

File: package.json [hunk 8]
```diff
@@ -11,7 +11,7 @@
     "start": "node ./dist/rpx-exui/api/server.bundle.js",
     "start:local": "cd api && yarn start",
     "build:node": "cd api && yarn build",
-    "build:prod": "ng build --prod",
+    "build:prod": "node --max_old_space_size=8192 node_modules/@angular/cli/bin/ng build --prod",
```
[the remark this answers is not in the document]
Comment (waynepHMCTS) [reply package.json:14]: Memory problem

File: src/hearings/hearings.routes.ts [hunk 9]
```diff
@@ -45,6 +46,22 @@ export const ROUTES: Routes = [
...
+        }
+      },
+      {
+        path: 'welsh-hearing',
+        component: WelshHearingComponent,
+        canActivate: [ HealthCheckGuard ],
+        data: {
+          title: 'HMCTS Manage cases | Request Hearing'
+        }
+      },
```
Comment (rayliangatos) [inline src/hearings/hearings.routes.ts:64]: merge issue, duplicated welsh-hearing?

File: src/hearings/hearings.routes.ts [hunk 10]
```diff
@@ -34,7 +35,7 @@ export const ROUTES: Routes = [
   },
   {
     path: 'request',
-    component: RequestHearingComponent,
+    component: null,
```
[the remark this answers is not in the document]
Comment (rayliangatos) [reply src/hearings/hearings.routes.ts:39]: it needs to be reverted to:
component: RequestHearingComponent,

File: src/hearings/containers/request-hearing/location-search-container/location-search-container.component.html [hunk 11]
```diff
@@ -0,0 +1,11 @@
+<exui-app-header></exui-app-header>
```
[the remark this answers is not in the document]
Comment (rayliangatos) [reply src/hearings/containers/request-hearing/location-search-container/location-search-container.component.html:1]: The exui-app-header line needs to be removed, as it has been defined in parent component.

**Review by rayliangatos — APPROVED** [review]

**Review by Kartheek-S — APPROVED** [review]

**Review by rayliangatos — APPROVED** [review]

**Review by Kartheek-S — APPROVED** [review]

**Pull request merged**